tasks/species_typing/task_tbprofiler.wdl

```diff
@@ -75,6 +88,92 @@ task tbprofiler {
+                    confidence.append("No WHO annotation")
+                  else:
+                    confidence.append(other_variant["annotation"][0]["who_confidence"])
+                except:
+                  confidence.append("No WHO annotation")
+              else:
+                confidence.append("No WHO annotation")
+            else:
+              if "annotation" in other_variant:  # check if who annotation field is present
+                for annotation in other_variant["annotation"]:
+                  if annotation["who_confidence"] != "Not assoc w R" or annotation["who_confidence"] != "":
+                    gene_name.append(other_variant["gene"])
+                    locus_tag.append(other_variant["locus_tag"])  
+                    variant_substitutions.append(other_variant["type"] + ":" + other_variant["nucleotide_change"] + "(" + other_variant["protein_change"] + ")")  # mutation_type:nt_sub(aa_sub)
+                    depth.append(other_variant["depth"])
+                    frequency.append(other_variant["freq"])
+                    confidence.append(annotation["who_confidence"])
+        
+        with open("GENE_NAME", "wt") as gene_name_fh:
+          gene_name_fh.write(','.join(gene_name))
+        
+        with open("LOCUS_TAG", "wt") as locus_tag_fh:
+          locus_tag_fh.write(','.join(locus_tag))
+        
+        with open("VARIANT_SUBSTITUTIONS", "wt") as variant_substitutions_fh:
+          variant_substitutions_fh.write(','.join(variant_substitutions))
+        
+        with open("OUTPUT_SEQ_METHOD_TYPE", "wt") as output_seq_method_type_fh:
+          output_seq_method_type_fh.write("~{output_seq_method_type}")
+
+        # file to be ingested into CDPH LIMS system
+        with open("tbprofiler_additional_outputs.csv", "wt") as additional_outputs_csv:
+          additional_outputs_csv.write("tbprofiler_gene_name,tbprofiler_locus_tag,tbprofiler_variant_substitutions,confidence,tbprofiler_output_seq_method_type\n")
+          additional_outputs_csv.write(";".join(gene_name) + "," + ";".join(locus_tag) + "," + ";".join(variant_substitutions) + ',' + ";".join(confidence) + ',' + "~{output_seq_method_type}")
+    
+        # laboratorian report
+        with open("tbprofiler_laboratorian_report.csv", "wt") as report_fh:
+          report_fh.write("tbprofiler_gene_name,tbprofiler_locus_tag,tbprofiler_variant_substitutions,confidence,depth,frequency,warning\n")
+          for i in range(0, len(gene_name)):
+            if not depth[i]:  # for cases when depth is null, it gets converted to 0
+              depth[i] = 0
+            warning = "Low depth coverage" if  depth[i] < int('~{min_depth}') else "" # warning when coverage is lower than the defined 'min_depth' times
+            report_fh.write(gene_name[i] + ',' + locus_tag[i] + ',' + variant_substitutions[i] + ',' + confidence[i] + ',' + str(depth[i]) + ',' + str(frequency[i]) + ',' + warning + '\n')
+
```

Comment:
This looks great. Noting this section for future reuse. As we add lims integration we'll want to generate files like these again. Nice work!!!